### PR TITLE
Fixes #8765: Add new password type 'pre-hashed' and 'script' 

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/8765

### CODE REVIEW

File: src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala [hunk 1]
```diff
@@ -199,7 +206,7 @@ object HashAlgoConstraint {
     }
 
     final case object Linux extends DerivedPasswordType {
-      final val name = "linux"
+      final val name = "Unix"
```
Comment (ncharles) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: doesn't it break something ?

File: src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala [hunk 2]
```diff
@@ -72,11 +72,20 @@ object HashAlgoConstraint {
...
   }
 
+  object PreHashed extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "pre-hashed"
+  }
+
+  object SCRIPT extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "script"
```
Comment (ncharles) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: doesn't it have other impact in term of code ?
what is it expected to do when it's a script ?

File: src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala (hunk 1 above)
Comment (VinceMacBuche) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: I kept the constructor that use 'Linux', name is only used a display

File: src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala (hunk 2 above)
Comment (VinceMacBuche) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: To evaluate this variable as a script. 

But the main goal is to provide a way to identify that the user would like to do a script, and ease it's usability

Comment (ncharles) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: but there is nothing in code to change behaviour if it's a script ...

Comment (VinceMacBuche) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: I'll let the evaluation do it for me. 

Not a problem to me if it does nothing but i can identify and help the user to enter a script

Comment (VinceMacBuche) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: Not a problem either if other hashes containing script are evalutated.

In fact that's a plain text that indicate that you want to save a script, but maybe it's not containing one.

In fact It does not indicate that the values should be evaluated as a script, but only that the user wanted to enter a script in there

Comment (ncharles) [inline src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala]: ok

**Pull request merged**